tests/artifacts_test.go

```diff
@@ -0,0 +1,207 @@
+//go:build test_integration
+
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	TestUtil "github.com/opendatahub-io/data-science-pipelines-operator/tests/util"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func (suite *IntegrationTestSuite) TestFetchArtifacts() {
+	suite.T().Run("Should successfully fetch and download artifacts", func(t *testing.T) {
+		log.Printf("Starting TestFetchArtifacts in namespace: %s", suite.DSPANamespace)
+
+		log.Println("Starting port-forwarding for artifact-service...")
+
+		minioNamespace := suite.DSPANamespace
+		minioServiceName := "minio-service" // Default for DSPA namespace
+
+		if suite.MinioNamespace != "default" {
+			minioNamespace = suite.MinioNamespace
+			minioServiceName = "minio" // Use "minio" for external namespace
+		}
+
+		// Run `kubectl get svc` to confirm MinIO service exists
+		cmdCheck := exec.Command("kubectl", "get", "svc", "-n", minioNamespace)
+		output, err := cmdCheck.CombinedOutput()
+		log.Printf("MinIO service check output:\n%s", string(output))
+		if err != nil {
+			log.Printf("Error checking MinIO service: %v", err)
+		}
+
+		// Start port-forwarding
+		cmd2 := exec.CommandContext(context.Background(),
+			"kubectl", "port-forward", "-n", minioNamespace, fmt.Sprintf("svc/%s", minioServiceName), fmt.Sprintf("%d:9000", 9000))
+		cmd2.Stderr = os.Stderr
+		cmd2.Stdout = os.Stdout
+		err = cmd2.Start()
+		require.NoError(t, err, "Failed to start port-forwarding")
+
+		go func() {
+			err := cmd2.Wait()
+			if err != nil {
+				log.Printf("The Minio port forward failed: %v\n", err)
+			}
+		}()
+
+		defer func() {
+			if cmd2.ProcessState == nil || !cmd2.ProcessState.Exited() {
+				_ = cmd2.Process.Kill()
+				cmd2.Wait()
+			}
+
+			log.Println("Port-forwarding process terminated.")
+		}()
+
+		// Wait briefly to ensure port-forwarding is established
+		time.Sleep(15 * time.Second)
```

Comment:
Rather than a sleep, could you please just verify that HTTP requests to Minio are working? If by 15 seconds it isn't working, then you can fail the test.

Reply:
Done, added a check for http requests to minio.